Update Code_Guidelines.rst (#358)
pyansys/PyAEDT: doc/source/Resources/Code_Guidelines.rst

~~~~~~~
Imports should always be placed at the top of the file, just after any
module comments and docstrings and before module globals and

> constants.  This reduces the likelihood of an import error that might only

Reviewer: I'd prefer keeping it as ``ImportError``.

> constants.  This reduces the likelihood of an import error that might only

Author: ```suggestion
constants.  This reduces the likelihood of an ``ImportError`` that might only
```

the doc special attribute of the object.

Write docstrings for all public modules, functions, classes, and
methods. Docstrings are not necessary for non-public methods, but such
methods should have comments that describe what they do.

To create a docstring, surround the comments with three double quotes
on either side.

> For a one-line docstring, keep both the starting and ending """ on the

Reviewer: ```suggestion
For a one-line docstring, keep both the starting and ending ``"""`` on the
```

same line. For example:

> 
> """This is a docstring.""".  

Reviewer: ```suggestion

.. code:: python

   """This is a docstring.""".  
```

> For a multi-line docstring, put the ending """ on a line by itself.

Reviewer: ```suggestion
For a multi-line docstring, put the ending ``"""`` on a line by itself.
```

Aside from the lack of comments, this complex nested validation method
is difficult to debug and validate with unit testing. It would

> be far better to implement more validation methods, join conditionals,

Author: ```suggestion
be far better to implement more validation methods and join conditionals.
```

> and so on.

Reviewer: ```suggestion
etc.
```

> and so on.

Author: ```suggestion
```

implement a unit test for ``is_consonant``.

PyAEDT-Specific Coding Guidelines

> -------------------------------

Reviewer: ```suggestion
---------------------------------
```

~~~~~~~~~~~~~~
PyAEDT has an internal logging tool named ``Messenger``
and a log file that is automatically generated in the project
folder. 

The following examples demonstrate how Messenger is used to 
write both to the internal AEDT message windows and the log file:

.. code:: python

    self.messenger.add_error_message("This is an error message.")
    self.messenger.add_warning_message("This is a warning message.")
    self.messenger.add_info_message("This is an info message.")

> These exmaples demonstratw how to to write messages only to the log file:

Reviewer: ```suggestion
These examples demonstrate how to to write messages only to the log file:
```

Hard-Coding Values
~~~~~~~~~~~~~~~~~~

> Do not write hard-coded values that the code is to
> use to the registry. Instead, use the Configuration service.

Reviewer: ```suggestion
Do not write hard-coded values to the registry. Instead, use the Configuration service.
```
